src/rotation.zig

```diff
@@ -357,35 +357,242 @@ pub const RotationModes = struct {
 			branchModels.deinit();
 		}
 
-		fn branchTransform(quad: *main.models.QuadInfo, data: BranchData) void {
-			for(&quad.corners) |*corner| {
-				if((!data.isConnected(Neighbor.dirNegX) and corner[0] == 0) or
-					(!data.isConnected(Neighbor.dirPosX) and corner[0] == 1) or
-					(!data.isConnected(Neighbor.dirNegY) and corner[1] == 0) or
-					(!data.isConnected(Neighbor.dirPosY) and corner[1] == 1) or
-					(!data.isConnected(Neighbor.dirDown) and corner[2] == 0) or
-					(!data.isConnected(Neighbor.dirUp) and corner[2] == 1)) return degenerateQuad(quad);
+		const Direction = enum(u2) {
+			negYDir = 0,
+			posXDir = 1,
+			posYDir = 2,
+			negXDir = 3,
+		};
+
+		const Pattern = union(enum) {
+			dot: void,
+			halfLine: struct {
+				dir: Direction,
+			},
+			line: struct {
+				dir: Direction,
+			},
+			bend: struct {
+				dir: Direction,
+			},
+			intersection: struct {
+				dir: Direction,
+			},
+			cross: void,
+		};
+
+		fn rotateQuad(originalCorners: [4]Vec2f, pattern: Pattern, minF: f32, maxF: f32, side: Neighbor) main.models.QuadInfo {
+			var corners: [4]Vec2f = originalCorners;
+
+			switch(pattern) {
+				.dot, .cross => {},
+				inline else => |typ| {
+					const angle: f32 = @as(f32, @floatFromInt(@intFromEnum(typ.dir)))*std.math.pi/2.0;
+					corners = .{
+						vec.rotate2d(originalCorners[0], angle, @splat(0.5)),
+						vec.rotate2d(originalCorners[1], angle, @splat(0.5)),
+						vec.rotate2d(originalCorners[2], angle, @splat(0.5)),
+						vec.rotate2d(originalCorners[3], angle, @splat(0.5)),
+					};
+				},
 			}
-		}
 
-		fn degenerateQuad(quad: *main.models.QuadInfo) void {
-			for(&quad.corners) |*corner| {
-				corner.* = @splat(0.5);
+			const offX: f32 = @floatFromInt(@intFromBool(@reduce(.Add, side.textureX()) < 0));
+			const offY: f32 = @floatFromInt(@intFromBool(@reduce(.Add, side.textureY()) < 0));
+
+			const norm: Vec3f = @floatFromInt(side.relPos());
+
+			const corns = .{
```

Comment:
I don't like that you use so many unusual abbreviations here, `corns` (🌽) `norm`, `tex`.
Here you could just inline `norm` and `tex` into there single use case. Same with `res`, why not just return directly?
As for 🌽s, I'd rename the above variable to corners2d and then you don't have the naming conflict.